### PR TITLE
Add HTTP response code to Spring WebFlux transactions

### LINKED ISSUE #2835: Normalize `http.response.status_code` for Metrics & Alerts
### Description

Please see the project description: https://github.com/getsentry/team-sdks/issues/9

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Set HTTP status code on transactions

## :bulb: Motivation and Context

Fixes #2835 for server side after client side was already done in #2786

## :green_heart: How did you test it?
Manually + Unit Tests

## :pencil: Checklist

- [x] I reviewed the submitted code.
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt [hunk 1]
```diff
@@ -117,6 +120,8 @@ class SentryTracingFilterTest {
         verify(fixture.hub).captureTransaction(
             check {
                 assertThat(it.contexts.trace!!.status).isEqualTo(SpanStatus.INTERNAL_ERROR)
+                assertThat(it.extras?.get(SpanDataConvention.HTTP_STATUS_CODE_KEY)).isEqualTo(500)
```
Comment (adinauer) [inline sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt:123]: `data` of transactions lands in `extras`, is this OK?

Comment (stefanosiano) [reply sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt:123]: why does it land in `extras`? shouldn't it be available through `getData()` method?

Comment (adinauer) [reply sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt:123]: The following code seems to do this on purpose:

https://github.com/getsentry/sentry-java/blob/695d3a3acd91e7759da0699f9b5e004d945de53b/sentry/src/main/java/io/sentry/protocol/SentryTransaction.java#L95-L100

Comment (stefanosiano) [reply sentry-spring-jakarta/src/test/kotlin/io/sentry/spring/jakarta/tracing/SentryTracingFilterTest.kt:123]: oh, i see, i was missing it 😅 
in https://github.com/getsentry/sentry-java/pull/2786 and https://github.com/getsentry/team-sdks/issues/9 we talked about span data, not transaction extra.
AFAIK the goal is to add the data directly to the spans, not to transactions.
Maybe @smeubank can help us removing any doubt?

**Review by cleptric — COMMENTED** [review]
For transactions, we should set the response status code on the event `context.response`. See https://github.com/getsentry/relay/blob/master/relay-general/src/protocol/contexts/response.rs

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Pull request merged**